### ISSUE TITLE
Support of Python3

### ISSUE THREAD

**Issue by PaxG** [nirs/slowfs#5]
Hello,

This software is very useful for testing purpose! But I would like to use it with Python3, and tried following code changes to be compatible Python 2 and 3:
exec f in d, d    =>    exec(f, d, d)
except socket.error, e:      =>    except socket.error as e:

But it does not work at this location:  exec(f, d, d)
TypeError: exec() arg 1 must be a string, bytes or code object
Debug shows the type of "f" object:
DEBUG: f=<_io.TextIOWrapper name='slowfs.cfg' mode='r' encoding='UTF-8'>

I tried with Python2 but could not get it work:
Traceback (most recent call last):
  File "slowfs.py", line 243, in <module>
    class SlowFS(fuse.Operations):
AttributeError: 'module' object has no attribute 'Operations'

Do we need a specific version of "fuse" library ?
Thanks !
Pascal

**Comment by nirs** [comment]
@PaxG, python 3 support is indeed a good idea, but I never checked if pyfuse works on python 3. Would you like to work on this?

Regarding python 2, it was tested on Fedora 25 with:
```
$ pip show fusepy
Name: fusepy
Version: 2.0.4
Summary: Simple ctypes bindings for FUSE
Home-page: http://github.com/terencehonles/fusepy
Author: Terence Honles
Author-email: terence@honles.com
License: ISC
Location: /usr/lib/python2.7/site-packages
Requires: 
```
```
$ python
Python 2.7.13 (default, Dec  1 2017, 09:21:53) 
[GCC 6.4.1 20170727 (Red Hat 6.4.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import fuse
>>> fuse.Operations
<class 'fuse.Operations'>
```

**Comment by PaxG** [comment]
Yes I would like to make it work with Python3, but not sure I've got the skills. I use:

```
% pip3.5 show fusepy
Name: fusepy
Version: 2.0.4
Summary: Simple ctypes bindings for FUSE
Home-page: http://github.com/terencehonles/fusepy
Author: Terence Honles
Author-email: terence@honles.com
License: ISC
...

```
```
python3.5 
Python 3.5.1 (default, May 27 2016, 05:38:51) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-4)] on linux
```
Code fixes for Python3 compatibility:
exec f in d, d      =>     exec(f,d,d)
except socket.error, e:    =>         except socket.error as e:
No other code fix found by "2to3" conversion command.

I still get this error:
  File "slowfs.py", line 68, in _load
    exec(f,d,d)
TypeError: exec() arg 1 must be a string, bytes or code object

What's the purpose of exec(f,d,d) ?
Thanks,
Pascal

**Comment by nirs** [comment]
Hi @PaxG,

exec is used to load the configuration file, which is a python module.

This works in both python 2 and 3:
```
$ cat test.cfg
value = "it works!'
```
```
>>> import six
>>> with open("test.cfg", "rb") as f:
...     data = f.read()
...     d = {}
...     six.exec_(data, d)
... 
>>> d['value']
'it works!'
```

**Comment by PaxG** [comment]
Hello Nir,
Thanks for your advice to use the six module for Python2/3 compatibility, it works !
I tried to git push my version but it's not allowed, please find it attached below.
This version probably needs a validation under Python2, but it works under Python3.
Best Regards
Pascal

> 403 Forbidden while accessing https://github.com/nirs/slowfs.git/info/refs

[slowfs.py.gz](https://github.com/nirs/slowfs/files/1590832/slowfs.py.gz)

**Comment by PaxG** [comment]
Found how to fork/branch/commit/push/send a pull request!
https://github.com/nirs/slowfs/pull/6

**Comment by nirs** [comment]
Some issues fixed in 9e835eacc0e8596d725f9c1ac55072ae7f24b58e, but we need more work.

**Comment by nirs** [comment]
@PaxG would you review and test #7?
